src/controllers/addRecipeController.ts

```diff
@@ -0,0 +1,84 @@
+import { Request, Response } from "express";
+import getDatabase from "../Services/databaseConnector";
+import {
+  ingredientIdExists,
+  isIdValid,
+  numberLengthIsValid,
+  stringLengthIsValid,
+  userIdExists,
+} from "../Services/validators";
+import { Connection } from "promise-mysql";
+
+export async function addRecipe(req: Request, res: Response) {
+
+  try {
+    const db: Connection = await getDatabase();
+    const userId: number = Number(req.params.userId);
+
+    if(!isIdValid(userId)) {
+      res.status(400).json({
+        message: "Invalid user id",
+      });
+      return
+    }
+
+    if(!(await userIdExists(db, userId))) {
+      res.status(400).json({
+        message: "Invalid user id",
+      });
+      return
+    }
```

Comment:
Can combine these 
```suggestion
    if(!isIdValid(userId) || !(await userIdExists(db, userId))) {
      res.status(400).json({
        message: "Invalid user id",
      });
      return
    }
```

If it's not an id that would work it stops checking at the `||`, so we won't get that SQL error any more, while still not duplicating code.

Reply:
done